### PR TITLE
Using surefire `rerunFailingTestsCount` for flaky tests

### PR DESCRIPTION
**Pull request by eyalkoren**
Closes #2554

### CODE REVIEW

File: apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/CallTreeTest.java [hunk 1]
```diff
@@ -59,6 +59,7 @@
 import static org.mockito.Mockito.doReturn;
 
 @DisabledOnOs(OS.WINDOWS)
+@DisabledOnAppleSilicon
```
Comment (eyalkoren) [inline apm-agent-plugins/apm-profiling-plugin/src/test/java/co/elastic/apm/agent/profiler/CallTreeTest.java:62]: Unrelated- was removed accidentally in #2686

File: pom.xml [hunk 2]
```diff
@@ -414,6 +414,7 @@
                 <configuration>
                     <enableAssertions>true</enableAssertions>
                     <trimStackTrace>false</trimStackTrace>
+                    <rerunFailingTestsCount>3</rerunFailingTestsCount>
```
Comment (JonasKunz) [inline pom.xml:417]: We should add the same configuration for the integration-tests below in the `maven-failsafe-plugin`, shouldn't we?
Link to docs: https://maven.apache.org/surefire/maven-failsafe-plugin/examples/rerun-failing-tests.html

**Comment by eyalkoren** [comment]
Since we are in the midst of the transition to using GH actions for reporting as well, the aggregation and report of such flaky tests results is out of scope in this PR
The only place I currently know this info is available is within the surefire output xml within a `<flakyFailure>` tag, but this means parsing all test output files. 
Example output for the `FlakyTest` added within this PR:
```
<?xml version="1.0" encoding="UTF-8"?>
<testsuite ...>
  ...
  <testcase name="passOnSecondRun" classname="co.elastic.apm.agent.testutils.FlakyTest" time="0">
    <flakyFailure message="[This test is expected to fail in its first run] &#10;Expecting actual not to be null" type="java.lang.AssertionError">
      <stackTrace><![CDATA[java.lang.AssertionError: 
      ...
      </stackTrace>
    </flakyFailure>
  </testcase>
</testsuite>
```
Another option is using something like [maven-surefire-report-plugin](https://maven.apache.org/surefire/maven-surefire-report-plugin/usage.html), but as far as I can see this only creates HTML reports. If that is the case, we can probably create it, archive it and add a link to it, but we will still want to have some kind of indication within the GitHub PR page if there were flaky tests.

**Review by JonasKunz — APPROVED** [review]
So build wise it seems to work as intended:
```
[2023-02-08T07:29:15.204Z] [INFO] Results:
[2023-02-08T07:29:15.204Z] [INFO] 
[2023-02-08T07:29:15.204Z] [WARNING] Flakes: 
[2023-02-08T07:29:15.204Z] [WARNING] co.elastic.apm.agent.testutils.FlakyTest.passOnSecondRun
[2023-02-08T07:29:15.204Z] [ERROR]   Run 1: FlakyTest.passOnSecondRun:38 [This test is expected to fail in its first run] 
[2023-02-08T07:29:15.204Z] Expecting actual not to be null
[2023-02-08T07:29:15.204Z] [INFO]   Run 2: PASS
[2023-02-08T07:29:15.204Z] [INFO] 
[2023-02-08T07:29:15.204Z] [INFO] 
[2023-02-08T07:29:15.204Z] [WARNING] Tests run: 970, Failures: 0, Errors: 0, Skipped: 1, Flakes: 1
```
We'll just need to figure out a way to make it show up in CI-report and/or GH comment.

EDIT: Looks like we responded at the same time, aggree that fixing the reporting is out of scope for this PR.

**Pull request merged**